src/state/slices/time_slice.ts

```diff
@@ -48,17 +57,24 @@ export const createTimeSlice: StateCreator<TimeSlice & DatasetSlice, [], [], Tim
     () => get().currentFrame,
     async (frame) => {
       set({ pendingFrame: frame });
-      await get().loadFrameCallback(frame);
-      set({ currentFrame: frame });
+      const result = await get().frameLoadCallback(frame);
+      if (result !== null) {
+        set({ currentFrame: frame });
+      } else if (get().pendingFrame === frame) {
+        // Reset pendingFrame if it hasn't changed. (e.g. no other calls to
+        // setFrame were made while this one was loading)
+        // TODO: More robust handling for requests? Request IDs?
+        set({ pendingFrame: get().currentFrame });
+      }
     }
   ),
-  loadFrameCallback: (_frame: number) => {
-    return Promise.resolve();
+  frameLoadCallback: (frame: number): Promise<FrameLoadResult> => {
+    return Promise.resolve({ frame, isFrameLoaded: false, isBackdropLoaded: false, backdropKey: null });
```

Comment:
Fake placeholder, which is expected to be changed immediately on store init in the actual app. I had it return a FrameLoadResult to keep the unit tests working with minimal changes.